aclmapping/tokenfactory/mappings.go

```diff
@@ -7,146 +7,203 @@ import (
+			ResourceType:       sdkacltypes.ResourceType_KV_TOKENFACTORY_DENOM,
+			IdentifierTemplate: string(tokenfactoryDenomKey),
 		},
 
 		// Gets Module Account information
 		{
 			AccessType:         sdkacltypes.AccessType_READ,
-			ResourceType:       sdkacltypes.ResourceType_KV,
-			IdentifierTemplate: utils.GetIdentifierTemplatePerModule(utils.AUTH, tokenfactorymoduletypes.ModuleName),
+			ResourceType:       sdkacltypes.ResourceType_KV_AUTH_ADDRESS_STORE,
+			IdentifierTemplate: string(authtypes.AddressStoreKey(moduleAdr)),
+		},
+		{
+			AccessType:         sdkacltypes.AccessType_READ,
+			ResourceType:       sdkacltypes.ResourceType_KV_BANK_BALANCES,
+			IdentifierTemplate: string(banktypes.CreateAccountBalancesPrefix(moduleAdr)),
 		},
 
-		// Sends coins to module account - deferred deposit
+		// Deposit into Sender's Bank Balance
+		{
+			AccessType:         sdkacltypes.AccessType_READ,
+			ResourceType:       sdkacltypes.ResourceType_KV_BANK_BALANCES,
+			IdentifierTemplate: string(banktypes.CreateAccountBalancesPrefixFromBech32(mintMsg.GetSender())),
+		},
+		{
+			AccessType:         sdkacltypes.AccessType_WRITE,
+			ResourceType:       sdkacltypes.ResourceType_KV_BANK_BALANCES,
+			IdentifierTemplate: string(banktypes.CreateAccountBalancesPrefixFromBech32(mintMsg.GetSender())),
+		},
 
-		// Updates Supply of the denom
+		// Read and update supply after burn
 		{
 			AccessType:         sdkacltypes.AccessType_READ,
-			ResourceType:       sdkacltypes.ResourceType_KV_BANK,
-			IdentifierTemplate: utils.GetPrefixedIdentifierTemplatePerModule(utils.BANK, "supply", denom),
+			ResourceType:       sdkacltypes.ResourceType_KV_BANK_SUPPLY,
+			IdentifierTemplate: supplyKey,
 		},
 		{
 			AccessType:         sdkacltypes.AccessType_WRITE,
-			ResourceType:       sdkacltypes.ResourceType_KV_BANK,
-			IdentifierTemplate: utils.GetPrefixedIdentifierTemplatePerModule(utils.BANK, "supply", denom),
+			ResourceType:       sdkacltypes.ResourceType_KV_BANK_SUPPLY,
+			IdentifierTemplate: supplyKey,
 		},
 
-		// Sends coins to the msgSender from the Module Account (deferred withdrawal)
 		{
 			AccessType:         sdkacltypes.AccessType_READ,
-			ResourceType:       sdkacltypes.ResourceType_KV_BANK,
-			IdentifierTemplate: utils.GetIdentifierTemplatePerModule(utils.BANK, mintMsg.Sender),
+			ResourceType:       sdkacltypes.ResourceType_KV_AUTH_ADDRESS_STORE,
+			IdentifierTemplate: string(authtypes.CreateAddressStoreKeyFromBech32(mintMsg.GetSender())),
 		},
 		{
 			AccessType:         sdkacltypes.AccessType_WRITE,
-			ResourceType:       sdkacltypes.ResourceType_KV_BANK,
-			IdentifierTemplate: utils.GetIdentifierTemplatePerModule(utils.BANK, mintMsg.Sender),
+			ResourceType:       sdkacltypes.ResourceType_KV_AUTH_ADDRESS_STORE,
+			IdentifierTemplate: string(authtypes.CreateAddressStoreKeyFromBech32(mintMsg.GetSender())),
+		},
+
+		// Coins removed from Module account (Deferred)
+
+		// Last Operation should always be a commit
+		{
```

Comment:
nit, we have a helper function called `CommitAccessOp()` that returns this access op